src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopy = Object.assign({}, state);
+  const newArr = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
+        newArr.push({ ...stateCopy });
+        break;
+      case 'removeProperties':
+        for (const keyToRemove of action.keysToRemove) {
+          delete stateCopy[keyToRemove];
+        }
+        newArr.push({ ...stateCopy });
+        break;
+
+      case 'clear':
+        for (const key in stateCopy) {
+          if (stateCopy.hasOwnProperty(key)) {
+            delete stateCopy[key];
+          }
+        }
+        newArr.push({ ...stateCopy });
+    }
```

Comment:
```suggestion
    }
    
    newArr.push({ ...stateCopy });
```
you could add 1 common push after switch and remove all from cases